src/platform/silabs/efr32/rs911x/BLEManagerImpl.cpp

```diff
@@ -0,0 +1,1008 @@
+#include <rsi_utils.h>
+#ifdef __cplusplus
+}
+#endif
+
+#include <ble/CHIPBleServiceData.h>
+#include <lib/support/CodeUtils.h>
+#include <lib/support/logging/CHIPLogging.h>
+#include <platform/CommissionableDataProvider.h>
+#include <platform/DeviceInstanceInfoProvider.h>
+#include <string.h>
+
+#if CHIP_ENABLE_ADDITIONAL_DATA_ADVERTISING
+#include <setup_payload/AdditionalDataPayloadGenerator.h>
+#endif
+
+extern uint16_t rsi_ble_measurement_hndl;
+extern rsi_ble_event_conn_status_t conn_event_to_app;
+extern sl_wfx_msg_t event_msg;
+
+StaticTask_t rsiBLETaskStruct;
+rsi_semaphore_handle_t sl_ble_init_sem;
+rsi_semaphore_handle_t sl_ble_event_sem;
+
+/* wfxRsi Task will use as its stack */
+StackType_t wfxBLETaskStack[WFX_RSI_TASK_SZ] = { 0 };
+
+using namespace ::chip;
+using namespace ::chip::Ble;
+
+void sl_ble_init()
+{
+    WFX_RSI_LOG("%s starting", __func__);
+
+    // registering the GAP callback functions
+    rsi_ble_gap_register_callbacks(NULL, NULL, rsi_ble_on_disconnect_event, NULL, NULL, NULL, rsi_ble_on_enhance_conn_status_event,
+                                   NULL, NULL, NULL);
+
+    // registering the GATT call back functions
+    rsi_ble_gatt_register_callbacks(NULL, NULL, NULL, NULL, NULL, NULL, NULL, rsi_ble_on_gatt_write_event, NULL, NULL, NULL,
+                                    rsi_ble_on_mtu_event, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL,
+                                    rsi_ble_on_event_indication_confirmation, NULL);
+
+    WFX_RSI_LOG("registering rsi_ble_add_service");
+
+    //  Exchange of GATT info with BLE stack
+    rsi_ble_add_matter_service();
+
+    //  initializing the application events map
+    rsi_ble_app_init_events();
+
+    WFX_RSI_LOG("StartAdvertising");
+    chip::DeviceLayer::Internal::BLEManagerImpl().StartAdvertising(); // TODO:: Called on after init of module
+    WFX_RSI_LOG("%s  Ended", __func__);
+}
+
+void sl_ble_event_handling_task(void)
+{
+    int32_t event_id;
+
+    WFX_RSI_LOG("%s starting", __func__);
+
+    //! This semaphore is waiting for wifi module initialization.
+    rsi_semaphore_wait(&sl_ble_init_sem, 0);
+
+    sl_ble_init();
+
+    // Application event map
+    while (1)
+    {
+        //! This semaphore is waiting for next ble event task
+        rsi_semaphore_wait(&sl_ble_event_sem, 0);
+
+        // checking for events list
+        event_id = rsi_ble_app_get_event();
+        switch (event_id)
+        {
+        case RSI_BLE_CONN_EVENT: {
+            rsi_ble_app_clear_event(RSI_BLE_CONN_EVENT);
+            chip::DeviceLayer::Internal::BLEMgrImpl().HandleConnectEvent();
+            WFX_RSI_LOG("%s Module got connected", __func__);
+        }
+        break;
+        case RSI_BLE_DISCONN_EVENT: {
+            // event invokes when disconnection was completed
+            WFX_RSI_LOG("%s Module got Disconnected", __func__);
+            chip::DeviceLayer::Internal::BLEMgrImpl().HandleConnectionCloseEvent(event_msg.reason);
```

Comment:
using the namespace allow you to reduce the function call lenght
`BLEMgrImpl().HandleConnectionCloseEvent`